### PR TITLE
Add activity handlers for some hacks

### LINKED ISSUE #30067: Electrohacked doors open at the start of the hack instead of the end.
**Describe the bug**  
When you electrohack a door open it takes about 30 seconds, but when you successfully do this the door opens and then 30 seconds past, giving a potential turret behind a door multiple turns to shoot you. 

**To Reproduce**  
1. Hack a door open with a turret behind it
2. Watch as the door opens and then you spend 30 seconds sitting there "hacking" while the turret shoots you for multiple turns.

**Expected behavior**  
To spend 30 seconds hacking, having the door open at the end and then possibly getting shot for one turn.

**Versions and configuration(please complete the following information):**
 - OS: [Windows 10]
 - Game Version: [0.D-2372-g1a0a791]
 - Graphics version: [Tiles]
 - Mods loaded: [StatsThroughSkills, Urban development,more locations, fuji's more buildings]

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
```SUMMARY: Bugfixes "Fix up front move cost for door and safe hacking"```

#### Purpose of change
Fixes #30067 

#### Describe the solution
Uses of iexamine::hack_attempt are replaced with activity handlers for hacking doors and safes.

#### Additional context
Fixing it for fuel terminal hacking is more complex because of the uistate data required. This PR doesn't attempt to handle that case.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -3549,3 +3552,138 @@ void activity_handlers::tree_communion_do_turn( player_activity *act, player *p
...
 }
+
+void activity_handlers::hack_door_finish( player_activity *act, player *p )
+{
+    if( p->has_trait( trait_id( "ILLITERATE" ) ) ) {
+        add_msg( _( "You can't read anything on the screen." ) );
+        act->set_to_null();
+        return;
+    }
+    bool using_electrohack = ( p->has_charges( "electrohack", 25 ) &&
```
Comment (Night-Pryanik) [inline src/activity_handlers.cpp]: External brackets are redundant, here and below in `using_fingerhack`.

**Comment by Night-Pryanik** [comment]
I see functions for hacking safes and doors are basically the same, the only differences are what terrain is set on success and message accompanying it. Consider writing a generic function for these two with terrain and message as arguments.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -5305,10 +5255,10 @@ hack_result iexamine::hack_attempt( player &p )
...
         return HACK_UNABLE;
     }
-    bool using_electrohack = ( p.has_charges( "electrohack", 25 ) &&
-                               query_yn( _( "Use electrohack?" ) ) );
-    bool using_fingerhack = ( !using_electrohack && p.has_bionic( bionic_id( "bio_fingerhack" ) ) &&
-                              p.power_level  > 24  && query_yn( _( "Use fingerhack?" ) ) );
+    bool using_electrohack = p.has_charges( "electrohack", 25 ) &&
+                             query_yn( _( "Use electrohack?" ) );
+    bool using_fingerhack = !using_electrohack && p.has_bionic( bionic_id( "bio_fingerhack" ) ) &&
+                            p.power_level  > 24  && query_yn( _( "Use fingerhack?" ) );
```
Comment (AMurkin) [inline src/iexamine.cpp]:
```suggestion
                            p.power_level > 24 && query_yn( _( "Use fingerhack?" ) );
```

File: src/activity_handlers.cpp [hunk 3]
```diff
@@ -3549,3 +3552,138 @@ void activity_handlers::tree_communion_do_turn( player_activity *act, player *p
...
+{
+    if( p->has_trait( trait_id( "ILLITERATE" ) ) ) {
+        add_msg( _( "You can't read anything on the screen." ) );
+        act->set_to_null();
+        return;
+    }
+    bool using_electrohack = p->has_charges( "electrohack", 25 ) &&
+                             query_yn( _( "Use electrohack?" ) );
+    bool using_fingerhack = !using_electrohack && p->has_bionic( bionic_id( "bio_fingerhack" ) ) &&
+                            p->power_level  > 24  && query_yn( _( "Use fingerhack?" ) );
```
Comment (AMurkin) [inline src/activity_handlers.cpp]:
```suggestion
                            p->power_level > 24 && query_yn( _( "Use fingerhack?" ) );
```

File: src/activity_handlers.cpp [hunk 4]
```diff
@@ -3549,3 +3552,138 @@ void activity_handlers::tree_communion_do_turn( player_activity *act, player *p
...
```
Comment (AMurkin) [inline src/activity_handlers.cpp]:
```suggestion
                            p->power_level > 24 && query_yn( _( "Use fingerhack?" ) );
```

**Pull request merged**